### PR TITLE
Parametrized configmaps

### PR DESCRIPTION
**Pull request by lemaitre-aneo**
This makes it possible to add extra configuration into all the ArmoniK configmaps. This is an easy way to adjust behavior of the deployment without needed to alter the terraform code. As an example, it integrates #834 for localhost deployment.

### CODE REVIEW

File: infrastructure/quick-deploy/aws/armonik/variables.tf [hunk 1]
```diff
@@ -54,14 +54,20 @@ variable "monitoring" {
...
-    max_polling_delay = string
-  })
+
+# Extra configuration
+variable "extra_conf" {
+  description = "Add extra configuration in the configmaps"
+  #type = object({
+  #  compute = optional(map(string), {})
+  #  control = optional(map(string), {})
+  #  core    = optional(map(string), {})
```
Comment (aneojgurhem) [inline infrastructure/quick-deploy/aws/armonik/variables.tf:64]: Is it intentional ?

File: infrastructure/quick-deploy/localhost/armonik/variables.tf [hunk 2]
```diff
@@ -47,14 +47,19 @@ variable "monitoring" {
...
-    min_polling_delay = string
-    max_polling_delay = string
-  })
+# Extra configuration
+variable "extra_conf" {
+  description = "Add extra configuration in the configmaps"
+  #type = object({
+  #  compute = optional(map(string), {})
+  #  control = optional(map(string), {})
+  #  core    = optional(map(string), {})
```
Comment (aneojgurhem) [inline infrastructure/quick-deploy/localhost/armonik/variables.tf:56]: idem

**Comment by aneojgurhem** [comment]
Good for me ! The easier the addition of envvars, the better !

File: infrastructure/quick-deploy/aws/armonik/variables.tf (hunk 1 above)
Comment (lemaitre-aneo) [reply infrastructure/quick-deploy/aws/armonik/variables.tf:64]: It is intentional: only the all-in-one deployments require Terraform 1.3, and can use optionals. The comment here is to document the format expected to "simulate" optionals in previous versions.

**Review by aneojgurhem — APPROVED** [review]

**Pull request merged**